test/web-platform-tests/to-upstream/html/semantics/embedded-content/the-img-element/remove-attribute-src.html

```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Removing src attribute from img</title>
+<link rel="author" title="atsikov" href="mailto:alexey.tsikov@gmail.com">
+<link rel=help href="https://html.spec.whatwg.org/#the-img-element">
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<body>
+<script>
+"use strict";
+
+async_test(t => {
+  const image = new window.Image();
+  image.onload = () => {
+    image.onload = t.unreached_func("onload should not be called with removed src");
+    image.onerror = t.unreached_func("onerror should not be called with removed src");
+    image.removeAttribute("src");
+
+    assert_equals(image.width, 0);
+    assert_equals(image.height, 0);
+    assert_equals(image.complete, false);
+    assert_equals(image.src, "");
+    assert_equals(image.currentSrc, "");
+
+    t.done();
```

Comment:
This is still not valid, because by calling the test done() here, you've made it impossible for image.onload/image.onerror to ever fire. (So the unreached_func assertions will never happen, even in a buggy implementation.)

In other words, I believe this test passes both with and without your change, making it a not-useful test.

Reply:
Without my changes `img.removeAttribute("src")` will raise an exception trying to load resource from `null` url (and that was the main reason for these changes), so test will fail even without any checks.

But you're right, since `onload` or `onerror` are async, it's better to leave `t.done()` in timeout only.

Reply:
I don't believe tests in the WPT framework fail due to raised exceptions; they fail due to assertions getting hit. (But I might be wrong; I haven't checked recently.)

Reply:
I tried and it failed with exception (in case canvas was installed, otherwise it didn't try to load anything).